src/Commands/Branch/DeleteCommand.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Pantheon\Terminus\Commands\Branch;
+
+use Pantheon\Terminus\Commands\TerminusCommand;
+use Pantheon\Terminus\Site\SiteAwareInterface;
+use Pantheon\Terminus\Site\SiteAwareTrait;
+use Terminus\Exceptions\TerminusException;
+
+class DeleteCommand extends TerminusCommand implements SiteAwareInterface
+{
+    use SiteAwareTrait;
+
+    /**
+     * Delete a git branch from a site.
+     *
+     * @command branch:delete
+     *
+     * @param string $branch_id The name of the branch to delete
+     * @param string $site_id The name of the site to delete the branch from.
+     * @throws \Terminus\Exceptions\TerminusException
+     */
+    public function deleteBranch($branch_id, $site_id)
```

Comment:
I'm amenable to changing, but the other commands would need to be updated to reflect this.